### DISCUSSION TITLE
im unable to get my previous data after migrating to quay.io/keycloak/keycloak:24.0.2

### DISCUSSION THREAD

**Discussion by SufiyanStark** [keycloak/keycloak#28351] [Q&A]
I recently transitioned from using jboss/keycloak to bitnami/keycloak and successfully migrated all my data, including users and tokens. However, upon migrating to [Quay](http://quay.io/keycloak/keycloak), I’m facing difficulties in retaining my data.

Here’s an overview of my setup:

I’m using a Docker Compose configuration where adi-auth service runs Keycloak ([Quay](http://quay.io/keycloak/keycloak) image), and I’ve specified environment variables for PostgreSQL database connection and admin credentials. Additionally, I’m attempting to import data from a JSON file during startup.

version: '2.4'
services:
  adi-auth:
    image: quay.io/keycloak/keycloak
    ports:
      - '8080:8080'
      - '6379:6379'
    depends_on:
      adi-auth-db:
        condition: service_healthy
    environment:
      - KC_DB_VENDOR=postgres
      - KC_DB_ADDR=adi-auth-db
      - KC_DB_DATABASE=auth-db
      - KC_DB_USER=postgres
      - KC_DB_SCHEMA=public
      - KC_DB_PASSWORD=####
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=admin
      - 'JAVA_OPTS_APPEND=-Dkeycloak.profile.feature.upload_scripts=enabled -Dkeycloak.profile.feature.token_exchange=enabled -Dkeycloak.profile.feature.admin_fine_grained_authz=enabled'
    healthcheck:
      test: ['CMD', 'curl', '-f', 'http://localhost:9990']
      interval: 30s
      timeout: 10s
      retries: 3
    volumes:
      - ./keycloak/realm-configs/realm.json:/opt/keycloak/data/import/realm.json
    command: ['start-dev', '--import-realm']
  adi-auth-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=#######
      - POSTGRES_DB=auth-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - ./postgres-auth-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8000:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  adi-backend-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=##########
      - POSTGRES_DB=adi-backend-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    # command: ["postgres", "-c", "log_statement=all"]
    volumes:
      - ./postgres-api-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8001:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  # Mail trap
  mail:
    image: mailhog/mailhog
    ports:
      - 8025:8025
      - 1025:1025strong text
issue I'm encountering is that although the setup seems correct, I'm unable to see the data that was previously available in bitnami/keycloak. Specifically, I'm missing user accounts, tokens, and other related data.

Could someone guide me on how to properly migrate data from bitnami/keycloak to quay.io/keycloak/keycloak? Additionally, are there any specific steps or considerations I should be aware of during this migration process?

Your insights and assistance would be greatly appreciated. Thank you!

![Screenshot (416)](https://github.com/keycloak/keycloak/assets/106228585/c771c087-8577-4c46-9d63-c68906ae8e1f)

**Comment by vmuzikar** [comment]
@SufiyanStark Hello, we do not deliver or maintain the Bitnami Keycloak, I'd suggest raising this issue with them.

**Reply by SufiyanStark** [reply]
im using quay.io/keycloak/keycloak:24.0.2 i was previously using bitnami and was able to see all my data but when i migrated to quay.io/keycloak/keycloak:24.0.2 im unable to get the same data that is the main issue

**Reply by vmuzikar** [reply]
Thank you for providing more context. Unfortunately, I'm not familiar with Bitnami Keycloak enough to understand what changes they made that might've caused this behaviour.

**Reply by SufiyanStark** [reply]
its related to environment variables thats what came to my understanding so currently I modified the code but now im getting the loading screen when i run the localhost:8080 in my browser and its stuck on that for arround 5min 
version: '2.4'
services:
  adi-auth:
    image: quay.io/keycloak/keycloak:24.0.2
    ports:
      - '8080:8080'
      - '6379:6379'
    depends_on:
      adi-auth-db:
        condition: service_healthy
    environment:
       - KC_METRICS_ENABLED=true
       - KC_HEALTH_ENABLED=true
       - KC_PROXY=edge
       - KC_HTTP_ENABLED=true
       - KC_HOSTNAME_STRICT_HTTPS=false
       - KC_DB=postgres
       - KC_DB_PASSWORD=auth-pass
       - KC_DB_URL=jdbc:postgresql://adi-auth-db/auth-db
       - KC_DB_USERNAME=postgres
       - KC_HOSTNAME=localhost
       - KEYCLOAK_ADMIN=keycloak-user
       - KEYCLOAK_ADMIN_PASSWORD=keycloak-pass
       - KC_FEATURES=token-exchange,scripts,admin-fine-grained-authz
    healthcheck:
      test: ['CMD', 'curl', '-f', 'http://localhost:9990']
      interval: 30s
      timeout: 10s
      retries: 3
    command: ['start-dev', '--import-realm']
    volumes:
      - ./keycloak/realm-configs/realm.json:/opt/keycloak/data/import/realm.json
  adi-auth-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=auth-pass
      - POSTGRES_DB=auth-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - ./postgres-auth-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8000:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  adi-backend-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=backend-pass
      - POSTGRES_DB=adi-backend-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    # command: ["postgres", "-c", "log_statement=all"]
    volumes:
      - ./postgres-api-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8001:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  # Mail trap
  mail:
    image: mailhog/mailhog
    ports:
      - 8025:8025
      - 1025:1025

**Reply by mabartos** [reply]
> healthcheck:
test: ['CMD', 'curl', '-f', 'http://localhost:9990']

@SufiyanStark For Keycloak 24, there's no port 9990. It was present in the legacy Wildfly-based distribution. The default ports are 8080(HTTP) and 8443(HTTPS).

**Reply by SufiyanStark** [reply]
@mabartos thank you i have changed it to 8443 now but still when i run localhost:8080 on browser im stuck on loading screen it says loading admin ui 

ill paste the code below for reference

version: '2.4'
services:
  adi-auth:
    image: quay.io/keycloak/keycloak:24.0.2
    ports:
      - '8080:8080'
      - '6379:6379'
    depends_on:
      adi-auth-db:
        condition: service_healthy
    environment:
       - KC_METRICS_ENABLED=true
       - KC_HEALTH_ENABLED=true
       - KC_PROXY=edge
       - KC_HTTP_ENABLED=true
       - KC_HOSTNAME_STRICT_HTTPS=false
       - KC_DB=postgres
       - KC_DB_PASSWORD=auth-pass
       - KC_DB_URL=jdbc:postgresql://adi-auth-db/auth-db
       - KC_DB_USERNAME=postgres
       - KC_HOSTNAME=localhost
       - KEYCLOAK_ADMIN=keycloak-user
       - KEYCLOAK_ADMIN_PASSWORD=keycloak-pass
       - KC_FEATURES=token-exchange,scripts,admin-fine-grained-authz
    healthcheck:
      test: ['CMD', 'curl', '-f', 'http://localhost:8443']
      interval: 30s
      timeout: 10s
      retries: 3
    command: ['start-dev', '--import-realm']
    volumes:
      - ./keycloak/realm-configs/realm.json:/opt/keycloak/data/import/realm.json
  adi-auth-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=auth-pass
      - POSTGRES_DB=auth-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - ./postgres-auth-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8000:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  adi-backend-db:
    image: postgres:13.1-alpine
    environment:
      - POSTGRES_PASSWORD=backend-pass
      - POSTGRES_DB=adi-backend-db
      - PGDATA=/var/lib/postgresql/data/pgdata
    # command: ["postgres", "-c", "log_statement=all"]
    volumes:
      - ./postgres-api-data:/var/lib/postgresql/data/pgdata
    ports:
      - '8001:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5
  # Mail trap
  mail:
    image: mailhog/mailhog
    ports:
      - 8025:8025
      - 1025:1025

**Comment by ThoreKr** [comment]
this is a cross-post from https://keycloak.discourse.group/t/unable-to-migrate-existing-data-to-the-quay-io-keycloak-keycloak-24-0-2/25346

formatting is unfortunately also unavailable on discourse.

**Comment by SufiyanStark** [comment]
@mabartos mabartos
im getting a container unhealthy when i run cypress test on my local but when i check docker desktop the container is running without any errors.
 If you have any additional insights or suggestions based on your experience, I would greatly appreciate them.

 the container runs fine even the keycloak is running but when i run the cypress test the same container becomes unhealthy and the test stops 
 
currently I have configured the code this way 

 environment:
      - KC_METRICS_ENABLED:true
      - KC_HEALTH_ENABLED:true
      - KC_PROXY:edge
      - KC_HTTP_ENABLED:true
      - KC_HOSTNAME_STRICT_HTTPS:false
      - KC_DB=postgres
      - KC_DB_URL=jdbc:postgresql://adi-auth-db/auth-db
      - KC_DB_USERNAME=xxx
      - KC_DB_PASSWORD=xxxxx
      - KEYCLOAK_ADMIN=xxxx
      - KEYCLOAK_ADMIN_PASSWORD=xxxxx
      - KC_FEATURES=token-exchange,scripts,admin-fine-grained-authz
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8080"]
      interval: 30s
      timeout: 10s
      retries: 3
    command: ['start-dev']

**Reply by mabartos** [reply]
Sorry, it's hard to say what can be wrong as not much information has been provided. From the first look, there's nothing I'd see strictly wrong that could cause the unhealthiness. 

Are there any different Keycloak options set for the Cypress tests? During the test, is Keycloak accessible on the `http://localhost:8080`? Or is there a specified HTTP relative path? Like `/auth` or something. You should check the Health endpoint provided by Keycloak on the website.

**Reply by SufiyanStark** [reply]
/auth was there before but now i have removed it 
actully i have dependencies like this in my code 
depends_on:
       xxx-auth-db:
         condition: service_healthy
 
i had to comment this one out so that it could work  
the tests are moving forward now but still the container is unhealthy

**Reply by ThoreKr** [reply]
Please stop crossposting, you are wasting other peoples time that way. Also, please, put your config in codeblocks. This unformatted mess is a horror to read and can't even be used to try to reproduce your issue.

What does your cypress test do? How do these issues manifest (what is cypress logging, does the container stop? what is the exit code?)
